src/components/RefinementList/RefinementList.tsx

```diff
@@ -1,37 +1,113 @@
 /** @jsx h */
 
-import { h, Component } from 'preact';
-import PropTypes from 'prop-types';
+import { h, createRef, Component } from 'preact';
 import cx from 'classnames';
 import { isSpecialClick, isEqual } from '../../lib/utils';
+import { PreparedTemplateProps } from '../../lib/utils/prepareTemplateProps';
 import Template from '../Template/Template';
 import RefinementListItem from './RefinementListItem';
 import SearchBox from '../SearchBox/SearchBox';
+import { RefinementListItem as TRefinementListItem } from '../../connectors/refinement-list/connectRefinementList';
+import { RefinementListOwnTemplates } from '../../widgets/refinement-list/refinement-list';
+import { RatingMenuTemplates } from '../../widgets/rating-menu/rating-menu';
+import { MenuTemplates } from '../../widgets/menu/menu';
+import { SearchBoxTemplates } from '../../widgets/search-box/types';
+import { CreateURL } from '../../types';
+
+export type RefinementListCSSClasses = {
+  depth?: string;
+  root?: string;
+  noRefinementRoot?: string;
+  list?: string;
+  item?: string;
+  selectedItem?: string;
+  parentItem?: string;
+  childList?: string;
+  searchBox?: string;
+  label?: string;
+  checkbox?: string;
+  labelText?: string;
+  count?: string;
+  noResults?: string;
+  showMore?: string;
+  disabledShowMore?: string;
+  disabledItem?: string;
+  searchable?: {
+    root: string;
+    form: string;
+    input: string;
+    submit: string;
+    submitIcon: string;
+    reset: string;
+    resetIcon: string;
+    loadingIndicator: string;
+    loadingIcon: string;
+  };
+};
+
+export type RefinementListProps = {
+  createURL: CreateURL<string>;
+  cssClasses?: RefinementListCSSClasses;
+  depth?: number;
+  facetValues?: TRefinementListItem[];
+  attribute?: string;
+  templateProps?:
+    | PreparedTemplateProps<RefinementListOwnTemplates>
+    | PreparedTemplateProps<RatingMenuTemplates>
+    | PreparedTemplateProps<MenuTemplates>;
```

Comment:
could the props & component be generic so that the templates type is passed as usage, and we don't need to expose RefinementListOwnTemplates

Reply:
Done in https://github.com/algolia/instantsearch.js/commit/232974f3ac03ccb3d76852fdb56c53062854b81e